File: js/stage1/S1_IF_03.js
// S1_IF_03 — Truthy/falsy guard
function normalizeName(input) {
  if (!input) return "Anonymous";
  return input.trim() || "Anonymous";
}
// Tests
["", " ", null, " Ola "].forEach((v) =>
  console.log(JSON.stringify(v), "=>", normalizeName(v))
);

File: Python/stage1/S1_IF_03.py
# S1_IF_03 — Normalize user name

def normalize_name(x):

    if not x:

        return "Anonymous"

    name = x.strip()

    if name == "":

        return "Anonymous"

    return name


# Tests

tests = ["", " ", None, " Ola "]

for t in tests:

    print(f"input={repr(t)} ->", normalize_name(t))
 